packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+
+module.exports = compile
+
+function compile (node) {
+  if (node === null || typeof node === 'number' || typeof node === 'boolean' || typeof node === 'string') {
+    return JSON.stringify(node)
+  }
+
+  const [type, value] = Object.entries(node)[0]
+
+  if (type === 'not') {
+    return `!(${compile(value)})`
+  } else if (type === 'isEmpty') {
+    return `(${compile(value)}).length === 0`
+  } else if (type === 'isDefined') {
+    return `(() => {
+        try {
+          ${value}
+          return true
+        } catch {
+          return false
+        }
+      })()`
+  } else if (type === 'instanceof') {
+    return `${compile(value[0])} instanceof ${value[1]}`
+  } else if (type === 'len' || type === 'count') {
+    return `(${compile(value)}).length`
```

Comment:
1. does this also works for Map? Sets? I think we need to call size on those.
2. do we need to use throwOnSideEfffects if length is a getter?

Reply:
The array `length` property cannot be redefined as a function or getter, so we should be safe there. However, maps and sets can have their `size` property replaced with a getter, so we should guard against that as well as supporting those with the `len`/`count` operation. Good call!

Reply:
Done 👍